src/Lykke.Service.CandlesHistory/Controllers/CandlesHistoryController.cs

```diff
@@ -43,21 +45,117 @@ public class CandlesHistoryController : Controller
+                // I.e., if we have 10 asset pairs, we get here 10 parallel tasks with 4 sequential
+                // data queries in each, but not 10 * 4 parallel tasks.
+                var result = await Task.WhenAll(
+                assetPairs
+                    .Where(p => _candleHistoryAssetConnections.ContainsKey(p.Id))
+                    .Select(async p => new CandlesHistoryDepthResponseModel
+                    {
+                        AssetPairId = p.Id,
+                        OldestAskTimestamp = await _candlesManager.TryGetOldestCandleTimestampAsync(p.Id, CandlePriceType.Ask, CandleTimeInterval.Sec),
+                        OldestBidTimestamp = await _candlesManager.TryGetOldestCandleTimestampAsync(p.Id, CandlePriceType.Bid, CandleTimeInterval.Sec),
+                        OldestMidTimestamp = await _candlesManager.TryGetOldestCandleTimestampAsync(p.Id, CandlePriceType.Mid, CandleTimeInterval.Sec),
+                        OldestTradesTimestamp = await _candlesManager.TryGetOldestCandleTimestampAsync(p.Id, CandlePriceType.Trades, CandleTimeInterval.Sec)
+
+                    })
+                );
+
+                return Ok(result);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ErrorResponse.Create("Internal", ex.Message));
+            }
+        }
+
+        /// <summary>
+        /// Shows history depth limits for the specified asset pair if it is supported.
+        /// </summary>
+        [HttpGet("availableAssetPairs/Depth/{assetPairId}")]
+        [SwaggerOperation("GetAssetPairHistoryDepth")]
+        [ProducesResponseType(typeof(CandlesHistoryDepthResponseModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.ServiceUnavailable)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        public async Task<IActionResult> GetAssetPairHistoryDepth(string assetPairId)
+        {
+            (var isOutOfService, var whatToSay) = CheckSelfState();
+            if (isOutOfService)
+                return whatToSay;
+
+            var resultTasks = Services.Candles.Constants.StoredPriceTypes
+                .Select(pt => _candlesManager.TryGetOldestCandleTimestampAsync(assetPairId, pt, CandleTimeInterval.Sec))
+                .ToList();
+
+            try
+            {
+                await Task.WhenAll(resultTasks);
+
+                // Actually, the tasks have been already awaited before. But we need to peek up their resulting values anyhow.
+                return Ok(new CandlesHistoryDepthResponseModel
+                {
+                    AssetPairId = assetPairId,
+                    OldestAskTimestamp = resultTasks[0].GetAwaiter().GetResult(),
```

Comment:
This is kinda fragile code. Consider to introduceTryGetOldestCandleAsync(), then FirstOrDefault(c => c.PriceType == XXX)?.TimeStamp